src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp

```diff
@@ -275,6 +275,9 @@ void DistributedSearchTreeImpl<DeviceType>::deviseStrategy(
     ExecutionSpace const &space, Predicates const &queries,
     DistributedTree const &tree, Indices &indices, Offset &offset, Distances &)
 {
+  Kokkos::Profiling::pushRegion(
+      "ArborX::DistributedSearchTree::deviseStrategy");
+
```

Comment:
I think there should be 4 regions in this function: overall, for top tree query, for computing new offset (parallel_for + scan), for truncation.